src/Collector/ClassNameCollector.php

```diff
@@ -33,9 +22,18 @@ public function satisfy(
         AstParserInterface $astParser
     ) {
         return preg_match(
-            '/'.$this->getRegexByConfiguration($configuration).'/i',
+            $this->getConfigurationRegex($configuration),
             $abstractClassReference->getClassName(),
             $collectorFactory
         );
     }
+
+    private function getConfigurationRegex(array $configuration)
```

Comment:
I think `ConfigurationRegex` makes more sense because it's the regex *of* `$configuration`. But I'm not a native english speaker so I may be wrong as well.

>what do you mean by "that's not a regex"? $configuration['regex']; is and was a regex?

I mean that the string return is not a complete regex, you have to add the delimiters behind, i.e. do `'/'.$this->getRegexByConfiguration().'/'` instead of just `$this->getRegexByConfiguration()`.